cmd/api/src/model/audit_test.go

```diff
@@ -0,0 +1,116 @@
+package model
+
+import (
+	"testing"
+
+	"github.com/gofrs/uuid"
+	"github.com/stretchr/testify/assert"
+)
+
+type MockAuditable struct {
+	Data AuditData
+}
+
+func (m MockAuditable) AuditData() AuditData {
+	return m.Data
+}
+
+func TestMatches_Success(t *testing.T) {
+	validEntry1 := AuditEntry{
+		CommitID: uuid.Must(uuid.NewV4()),
+		Action:   "action",
+		Model:    MockAuditable{AuditData{"key": "value"}},
+		Status:   "ok",
+		ErrorMsg: "err",
+	}
+
+	validEntry2 := AuditEntry{
+		CommitID: uuid.Must(uuid.NewV4()),
+		Action:   "action",
+		Model:    MockAuditable{AuditData{"key": "value"}},
+		Status:   "ok",
+		ErrorMsg: "err",
+	}
+
+	assert.True(t, validEntry1.Matches(validEntry2), "Expected validEntry1 to match validEntry2")
+}
+
+func TestMatches_FailureAction(t *testing.T) {
+	actionEntry1 := AuditEntry{
+		CommitID: uuid.Must(uuid.NewV4()),
+		Action:   "action",
+		Model:    MockAuditable{AuditData{"key": "value"}},
+		Status:   "ok",
+		ErrorMsg: "err",
+	}
+
+	actionEntry2 := AuditEntry{
+		CommitID: uuid.Must(uuid.NewV4()),
+		Action:   "invalid action",
+		Model:    MockAuditable{AuditData{"key": "value"}},
+		Status:   "ok",
+		ErrorMsg: "err",
+	}
+
+	assert.False(t, actionEntry1.Matches(actionEntry2), "Expected actionEntry1 to match actionEntry2")
```

Comment:
I think the error message here (and for the following tests that use `assert.False()`) might be flipped, since we are testing that the two entries do not match

Reply:
i think this is a copy pasta error 🫠 my bad.  i think that this should be false, but the message should be "expected entry1 not to match entry2".  unless i am misunderstanding the function, i think it is checking that all fields match besides the commitid. so the first test case is checking that scenario and then the others are checking for mismatches on the other fields.  or thats how i thought about it in my head 😅  let me know if that is not the case.

Reply:
That is my understanding too, the tests themselves all make sense to me!